### PR TITLE
Dp 3119 - Generate postgre sql ddl from excel

### PR DESCRIPTION
**Pull request by chlocarlos**

### CODE REVIEW

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/GeneratePostgreSqlDdlFromExcel.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import com.northconcepts.datapipeline.foundations.schema.EntityDef;
+import com.northconcepts.datapipeline.foundations.schema.SchemaDef;
+import com.northconcepts.datapipeline.foundations.tools.GenerateEntityFromDataset;
+import com.northconcepts.datapipeline.sql.postgresql.CreatePostgreSqlDdlFromSchemaDef;
+
+public class GeneratePostgreSqlDdlFromExcel {
+
+    public static void main(String[] args) {
+        ExcelDocument document = new ExcelDocument(ProviderType.POI_XSSF)
+                .open(new File("example/data/input/credit-balance-01.xls"));
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/GeneratePostgreSqlDdlFromExcel.java]: let's use a modern XLSX file that works with the default `ExcelDocument()` constructor.

**Review by DeleTaylor — CHANGES_REQUESTED** [review]

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/GeneratePostgreSqlDdlFromExcel.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        DataReader reader = new ExcelReader(document).setFieldNamesInFirstRow(true);
+
+        GenerateEntityFromDataset generator = new GenerateEntityFromDataset();
+        EntityDef entity = generator.generateEntity(reader).setName("creditBalance");
+
+        SchemaDef schema = new SchemaDef("credit-balance-01")
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/GeneratePostgreSqlDdlFromExcel.java]: Let's remove the dashes to use a schema name that's compatible with databases.

**Review by DeleTaylor — CHANGES_REQUESTED** [review]

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/GeneratePostgreSqlDdlFromExcel.java (hunk 1 above)
Comment (chlocarlos) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/GeneratePostgreSqlDdlFromExcel.java]: Done

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/GeneratePostgreSqlDdlFromExcel.java (hunk 2 above)
Comment (chlocarlos) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/GeneratePostgreSqlDdlFromExcel.java]: Done

**Review by DeleTaylor — APPROVED** [review]

**Pull request merged**